tracer/src/Datadog.Trace/PDBs/SourceLink/AzureDevOpsSourceLinkUrlParser.cs

```diff
@@ -64,6 +70,35 @@ internal override bool TryParseSourceLinkUrl(Uri uri, [NotNullWhen(true)] out st
         return false;
     }
 
+    private static string? BuildRepositoryUrl(Uri uri)
+    {
+        var segments = uri.AbsolutePath.Split(['/'], StringSplitOptions.RemoveEmptyEntries);
```

Comment:
nit: we have a `SpanCharSplitter` now which can avoid allocations in place of the "expensive" split

Reply:
I tend to say this is not worth it because:
1. AFAICT `SparCharSplitter` doesn't support an indexing operator, which means switching to it would require making this code less readable
2. This code path generally speaking will only occur once in the life time of the application, so I think adding this one allocation should be fine?

WDYT?

Reply:
Yeah, that's fine by me, makes sense 👍